bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/handler/Powerline546EHandler.java

```diff
@@ -74,41 +83,137 @@ public Powerline546EHandler(Bridge bridge, HttpClient httpClient,
+                updateThingChannelState(CHANNEL_VOLTAGE,
+                        new QuantityType<>(device.getPowermeter().getVoltage(), SmartHomeUnits.VOLT));
+            }
+            if (device.isSwitchableOutlet() && device.getSwitch() != null) {
+                updateThingChannelState(CHANNEL_MODE, new StringType(device.getSwitch().getMode()));
+                updateThingChannelState(CHANNEL_LOCKED,
+                        BigDecimal.ZERO.equals(device.getSwitch().getLock()) ? OpenClosedType.OPEN
+                                : OpenClosedType.CLOSED);
+                updateThingChannelState(CHANNEL_DEVICE_LOCKED,
+                        BigDecimal.ZERO.equals(device.getSwitch().getDevicelock()) ? OpenClosedType.OPEN
+                                : OpenClosedType.CLOSED);
+                if (device.getSwitch().getState() == null) {
+                    updateThingChannelState(CHANNEL_OUTLET, UnDefType.UNDEF);
+                } else {
+                    updateThingChannelState(CHANNEL_OUTLET,
+                            SwitchModel.ON.equals(device.getSwitch().getState()) ? OnOffType.ON : OnOffType.OFF);
+                }
             }
+        }
+    }
+
+    /**
+     * Updates thing properties.
+     *
+     * @param device the {@link AVMFritzBaseModel}
+     */
+    private void updateProperties(AVMFritzBaseModel device) {
+        Map<String, String> editProperties = editProperties();
+        editProperties.put(Thing.PROPERTY_FIRMWARE_VERSION, device.getFirmwareVersion());
+        updateProperties(editProperties);
+    }
+
+    /**
+     * Updates thing channels.
+     *
+     * @param channelId ID of the channel to be updated.
+     * @param state State to be set.
+     */
+    private void updateThingChannelState(String channelId, State state) {
+        Channel channel = thing.getChannel(channelId);
+        if (channel != null) {
+            updateState(channel.getUID(), state);
         } else {
-            setStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.GONE, "Device not present in response");
+            logger.debug("Channel '{}' in thing '{}' does not exist, recreating thing.", channelId, thing.getUID());
+            createChannel(channelId);
         }
-        super.addDeviceList(devicelist);
     }
 
     /**
-     * Updates things from device model.
+     * Creates new channels for the thing.
      *
-     * @param thing Thing to be updated.
-     * @param device Device model with new data.
+     * @param channelId ID of the channel to be created.
      */
+    private void createChannel(String channelId) {
+        ThingHandlerCallback callback = getCallback();
+        if (callback != null) {
+            ChannelUID channelUID = new ChannelUID(thing.getUID(), channelId);
+            ChannelTypeUID channelTypeUID = CHANNEL_BATTERY.equals(channelId)
+                    ? new ChannelTypeUID("system", "battery-level")
```

Comment:
I found the constant for this!
`DefaultSystemChannelTypeProvider.SYSTEM_CHANNEL_BATTERY_LEVEL.getUID()`

Reply:
Nice :wink:

To make you happy i submitted https://github.com/openhab/openhab-addons/pull/7650.